### ISSUE TITLE
pod-failure action throwing Forbidden error 

### ISSUE THREAD

**Issue by anuragpaliwal80** [chaos-mesh/chaos-mesh#1574]
## Bug Report

**What version of Kubernetes are you using?**
k8s version: 1.19

**What version of Chaos Mesh are you using?**

chaos mesh version: 1.0.1
 
**What did you do?**

Tried applying `pod-chaos` action.

**What did you expect to see?**
It should not have any error while describing the created PodChaos object.
**What did you see instead?**
```
Events:
  Type     Reason             Age                  From                 Message
  ----     ------             ----                 ----                 -------
  Warning  ChaosInjectFailed  39s (x15 over 2m2s)  podchaos-controller  Pod "dep1-5648b4dcbf-nm22p" is invalid: spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds` or `spec.tolerations` (only additions to existing tolerations)
  core.PodSpec{
    ... // 10 identical fields
    AutomountServiceAccountToken: nil,
    NodeName:                     "ip-10-79-146-75.ec2.internal",
    SecurityContext: &core.PodSecurityContext{
      ... // 11 identical fields
      FSGroupChangePolicy: nil,
      Sysctls:             nil,
-     SeccompProfile:      nil,
+     SeccompProfile:      &core.SeccompProfile{Type: "RuntimeDefault"},
    },
    ImagePullSecrets: nil,
    Hostname:         "",
    ... // 15 identical fields
  }
```
**Output of chaosctl**

**Comment by fewdan** [comment]
Thank you for receiving the issue you recorded. We will check this issue.

**Comment by cwen0** [comment]
@anuragpaliwal80 This behavior is very strange, `pod-failure` just updates the image of the target pod and adds a new annotation. https://github.com/chaos-mesh/chaos-mesh/blob/release-1.0/controllers/podchaos/podfailure/types.go#L199-L256

**Comment by YangKeao** [comment]
Do you have any webhook which will add `SeccompProfile` automatically? I don't think it has any connection with Chaos Mesh.

**Comment by anuragpaliwal80** [comment]
As per https://kubernetes.io/docs/tutorials/clusters/seccomp/ `Kubernetes lets you automatically apply seccomp profiles loaded onto a Node to your Pods and containers.`

The annotation getting applied on my service pod is:
```
annotations:
    seccomp.security.alpha.kubernetes.io/pod: runtime/default
```

**Comment by STRRL** [comment]
I think it might be related to the conversion with different versions of kube-client and kubernetes apiserver.

Using PATCH instead of UPDATE might resolve this problem without upgrading kubeclient.

**Comment by anuragpaliwal80** [comment]
Any idea, how should we proceed on this ?

**Comment by STRRL** [comment]
Sorry for the late response, I am going to fix that. :see_no_evil:

**Comment by 51n15t9r** [comment]
We ran into the exact same problem with version 1.21 of chaos-mesh. 
Is there a work-around we can use in the meantime?

**Comment by STRRL** [comment]
Hi @anuragpaliwal80, @51n15t9r sorry for the late response, we have fixed this issue at #2018 for v2.0 and #2026 for v1.2.

Please upgrade Chaos Mesh to v2.0.0-beta.2 or v1.2.2!

**Comment by anuragpaliwal80** [comment]
Closing this issue, will check with action in future.
